maintenance_equipment_contract/views/maintenance_equipment.xml

```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2019 Creu Blanca
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <record model="ir.ui.view" id="hr_equipment_view_form">
+        <field
+            name="name"
+        >maintenance.equipment.form (in maintenance_equipment_contract)</field>
+        <field name="model">maintenance.equipment</field>
+        <field name="inherit_id" ref="maintenance.hr_equipment_view_form" />
+        <field name="arch" type="xml">
+            <div name="button_box" position="inside">
+                <field name="contract_ids" invisible="1" />
+                <button
+                    name="action_view_contracts"
+                    class="oe_stat_button"
+                    icon="fa-usd"
+                    type="object"
+                >
+                    <field name="contract_count" widget="statinfo" string="Contracts" />
+                </button>
+            </div>
+        </field>
+    </record>
```

Comment:
Suggested improvement (not blocking): add a `search` view with filters "Has contracts" / "Has no contracts"

Reply:
Added them